tests/utils_tests/test_timesince.py

```diff
@@ -123,3 +123,8 @@ def test_thousand_years_ago(self):
         t = datetime.datetime(1007, 8, 14, 13, 46, 0)
         self.assertEqual(timesince(t, self.t), '1000\xa0years')
         self.assertEqual(timeuntil(self.t, t), '1000\xa0years')
+
+    def test_leap_year_new_years_eve(self):
```

Comment:
I don't see any test failures if your fix is removed. I think that you should add test like in [regression](https://code.djangoproject.com/ticket/27637#comment:6) but with specific values, e.g.:
```python
    def test_leap_year_new_years_eve(self):
        t = datetime.date(2016, 12, 31)
        now = datetime.datetime(2016, 12, 31, 18, 18, 0)
        self.assertEqual(timesince(t + self.oneday, now), '0\xa0minutes')
        self.assertEqual(timeuntil(t - self.oneday, now), '0\xa0minutes')
```